routes/user.go

```diff
@@ -4,41 +4,84 @@ import (
 	"encoding/json"
 	"github.com/HRODEV/project7_8/dbActions"
 	"github.com/HRODEV/project7_8/models"
-	"github.com/gorilla/mux"
+	"github.com/HRODEV/project7_8/services"
+	"github.com/dgrijalva/jwt-go"
 	_ "github.com/jinzhu/gorm/dialects/sqlite"
 	"io/ioutil"
 	"net/http"
-	"strconv"
+	"strings"
+	"time"
 )
 
 func UserAuthGet(w http.ResponseWriter, r *http.Request, utils Utils) interface{} {
-	http.Error(w, "Not implemented yet", http.StatusNotImplemented)
-	return nil
+	authorizationHeader := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
+
+	if len(authorizationHeader) != 2 || authorizationHeader[0] != "Basic" {
+		http.Error(w, "Authorization header not found or empty", http.StatusBadRequest)
+		return nil
+	}
+
+	authService := services.AuthService{}
+	userCredentials := strings.Split(authService.DecodeAuthorzationHeader(authorizationHeader[1]), ":")
+
+	// Verify email and password
+	user := models.User{}
+	utils.db.Where("email = ? AND password = ?", userCredentials[0], userCredentials[1]).Find(&user)
+
+	if user.ID == 0 {
+		http.Error(w, "Combination of username and password cannot be found", http.StatusUnauthorized)
+		return nil
+	}
+
+	// Create the token
+	token := jwt.New(jwt.SigningMethodHS256)
+
+	// Create a map to store our claims
+	claims := token.Claims.(jwt.MapClaims)
```

Comment:
Deze wordt gebruik in het token dat gegenereerd is. Dus de claim die in deze map komen worden aan het token toegevoegd. Dat zijn in dit geval `exp` waarin de tijd in de vorm van UNIX timestamp staat tot wanneer het token geldig is, en `sub` waarin het userID van de gebruiker staat.